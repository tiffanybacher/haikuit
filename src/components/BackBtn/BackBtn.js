import React from 'react';

const BackBtn = (props) => {
  return (
    <button 
      className="back-btn" 
      onClick={props.goBack}>
      Back
    </button>
  );
}

export default BackBtn;
