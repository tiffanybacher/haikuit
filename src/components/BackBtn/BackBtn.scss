.back-btn {
  color: black;
  font-style: italic;
  text-decoration: underline;
}