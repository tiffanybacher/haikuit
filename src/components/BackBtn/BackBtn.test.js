import React from 'react';
import { shallow } from 'enzyme';
import BackBtn from './BackBtn';

describe('BackBtn', () => {
  let goBack;
  let wrapper;

  beforeEach(() => {
    goBack = jest.fn();
    wrapper = shallow(<BackBtn goBack={goBack} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke goBack on click', () => {
    wrapper.find('.back-btn').simulate('click');

    expect(goBack).toHaveBeenCalled();
  });
});