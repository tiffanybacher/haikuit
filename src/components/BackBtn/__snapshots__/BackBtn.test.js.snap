// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackBtn should match snapshot 1`] = `
<button
  className="back-btn"
  onClick={[MockFunction]}
>
  Back
</button>
`;
